docs(make-your-components-compatible-with-aao): add Asset Description in How to improve the compatibility (#1032)
anatawa12/AvatarOptimizer: .docs/content/docs/developers/make-your-components-compatible-with-aao/index.md

Please remove your components before Avatar Optimizer processes as much as possible.
If you cannot remove some components, please register them to Avatar Optimizer.

> Since Avatar Optimizer v1.7.0, [Asset Description] is added for components that doesn't work on or after build time.

Reviewer: ```suggestion
Since Avatar Optimizer v1.7.0, [Asset Description] is added for components that doesn't process during build time or runtime.
```

Reviewer: なるほど(Asset Description側にduring buildとあったのでこうしてみてました)

Reviewer: ```suggestion
Since Avatar Optimizer v1.7.0, [Asset Description] is added for components that doesn't process on build or at runtime.
```

If your tool process nothing at build time, you can use this to register your components.

> If your tool process something at build time, 

Reviewer: ```suggestion
If your tool process something at build time, registering with Asset Description is not recommended.
```

> you should remove components on the build time instead of registering with Asset Description.

Reviewer: ```suggestion
```

> This prevents Avatar Optimizer from accidentally removing components and disabling your tool when the execution order is unexpected.

Reviewer: ```suggestion
This is to prevent Avatar Optimizer from accidentally removing components and causing your tool to not work properly when the processes execution order is unexpected/incorrect.
```

Author: this prevets~のほうが聞く気がする.

Reviewer: 前文全体をpreventsの主語にするにはどこかしっくり来ない気がして変えてました
This preventsでも大丈夫ではあると思います

Author: bdちゃんに聞いて変わらんとのことなのでThis preventsでいきます
